### PR TITLE
[Workspace] search use case overview page

### PR DESCRIPTION
**Pull request by Hailong-am**
### Description

Add search use case overview page
 
breadcrumbs will handled by other PR 

### Issues Resolved

#7807 

## Screenshot

image

## Changelog

- feat: [Workspace] Add search use case overview page

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/home/public/plugin.ts [hunk 1]
```diff
@@ -168,6 +174,36 @@ export class HomePublicPlugin
       workspaceAvailability: WorkspaceAvailability.outsideWorkspace,
     });
 
+    if (core.chrome.navGroup.getNavGroupEnabled()) {
```
Comment (SuZhou-Joe) [inline src/plugins/home/public/plugin.ts:177]: Nit: Not sure if it is suitable to register the page inside home plugin.

**Review by SuZhou-Joe — APPROVED** [review]

File: src/plugins/home/public/application/components/usecase_overview/search_use_case_setup.tsx [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+      description: i18n.translate('home.searchOverview.searchTypes.analyzer.description', {
+        defaultMessage:
+          'Analyzers prepare text for indexing. For example, HTML text typically has its tags removed. English text typically treats paint, painted, and painting as equivalent by mapping them all to the token paint. You can use OpenSearch’s extensive library of standard analyzers or create your own.',
+      }),
+      icon: searchIcon,
+      footer: (
+        <EuiLink
+          external
+          target="_blank"
+          href="https://opensearch.org/docs/latest/analyzers/search-analyzers/"
```
Comment (ruanyl) [inline src/plugins/home/public/application/components/usecase_overview/search_use_case_setup.tsx:157]: Shall we use/add these links to `doc_links_service` in `src/core/public/doc_links/doc_links_service.ts`?

**Review by ruanyl — APPROVED** [review]

File: src/plugins/home/public/plugin.ts (hunk 1 above)
Comment (Hailong-am) [reply src/plugins/home/public/plugin.ts:177]: agree, It's a little wired we put into home plugin, we might need to have a new plugin to store use case overview pages.

File: src/plugins/home/public/application/components/usecase_overview/search_use_case_setup.tsx (hunk 2 above)
Comment (Hailong-am) [reply src/plugins/home/public/application/components/usecase_overview/search_use_case_setup.tsx:157]: good suggestion, we can have a follow up PR to add those links into doc_link_services.

**Comment by Hailong-am** [comment]
backport depends on https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7884 to merged first

**Pull request merged**